src/lib/gadgets/gadgets.ts

```diff
@@ -314,6 +315,31 @@ const Gadgets = {
     return compactMultiRangeCheck(xy, z);
   },
 
+  /**
+   * Checks that three {@link Field} elements are less than 2^12 (using only one row) by verifying that they exist in the [RANGE_CHECK_TABLE](https://github.com/o1-labs/proof-systems/blob/master/kimchi/src/circuits/lookup/tables/mod.rs).
+   *
+   * It's possible to use this to check that a value is less than x bits by scaling the value by 2^(12 - x) before passing it in (with x < 12).
+   *
+   * You can find more details about lookups in the [Mina book](https://o1-labs.github.io/proof-systems/specs/kimchi.html?highlight=lookup%20gate#lookup)
+   *
+   * @param v0 - The first {@link Field} element to be checked.
+   * @param v1 - The second {@link Field} element to be checked.
+   * @param v2 - The third {@link Field} element to be checked.
+   *
+   * @throws Throws an error if one of the input values exceeds 2^12.
+   *
+   * @example
+   * ```typescript
+   * let a = Field(4000);
+   * let b = a.mul(2 ** 4); // scale `a` so we can check if it's less than 8 bits
+   * three12Bit(a, a, a); // works
+   * three12Bit(a, a, b); // throws an error since b is greater than 12 bits (and `a` is greater than 8 bits)
+   * ```
+   */
+  three12Bit(v0: Field, v1: Field, v2: Field) {
```

Comment:
can we name this function more similarly to other range check gadgets? we have `rangeCheck64` and `rangeCheck32`, as well as `multiRangeCheck` for the 3x88 bit check. this one could be `rangeCheck12x3` maybe? at least it should say "range check"